aopy/data/bmi3d.py

```diff
@@ -2242,6 +2242,85 @@ def tabulate_behavior_data_tracking_task(preproc_dir, subjects, ids, dates, meta
     df = pd.concat([df, new_df], ignore_index=True)
     return df
 
+def wrapper_tabulate_random(preproc_dir, subjects, ids, dates, metadata=[], 
+                                      df=None, event_code_type = 'code'):
+    '''
+    Wrapper around tabulate_behavior_data() specifically for random target location experiments. 
+    Uses the task event names (b'TARGET_ON' and b'TRIAL_END', specifically) or corresponding codes
+    to find start and end times for experiments.
+
+    Args:
+        preproc_dir (str): base directory where the files live
+        subjects (list of str): Subject name for each recording
+        ids (list of int): Block number of Task entry object for each recording
+        dates (list of str): Date for each recording
+        metadata (list, optional): list of metadata keys that should be included in the df
+        df (DataFrame, optional): pandas DataFrame object to append. Defaults to None.
+        event_code_type: (str, optional): type of event codes to use. Defaults to 'code'. Other
+        choice is 'event'.
+
+    Returns:
+        pd.DataFrame: pandas DataFrame containing the concatenated trial data with columns:
+            | **subject (str):** subject name
+            | **te_id (str):** task entry id
+            | **date (str):** date of recording
+            | **event_names (ntrial):** event name segments for each trial
+            | **event_times (ntrial):** time segments for each trial
+            | **%metadata_key% (ntrial):** requested metadata values for each key requested
+            | **target_idx (ntrial): ** target index for each trial within a unique data session
+            | **target_loc (ntrial): ** target locations (x,y,z) for each trial 
+'''
+    # Set up how to tabulate based on event_code_type.
+    if event_code_type == 'event':
+        trial_end_codes = [b'TRIAL_END']
+        trial_start_codes = [b'TARGET_ON']
+        reward_codes = [b'REWARD']
+        penalty_codes = [['HOLD_PENALTY'], ['DELAY_PENALTY']]
```

Comment:
should probably include timeout penalties in penalty_codes and pause events in trial_end_codes